public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -718,16 +742,23 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
   }
 
   getQueryDisplayText(query: TempoQuery) {
-    if (query.queryType === 'nativeSearch') {
-      let result = [];
-      for (const key of ['serviceName', 'spanName', 'search', 'minDuration', 'maxDuration', 'limit']) {
-        if (query.hasOwnProperty(key) && query[key as keyof TempoQuery]) {
-          result.push(`${startCase(key)}: ${query[key as keyof TempoQuery]}`);
-        }
-      }
-      return result.join(', ');
+    const validKeys = ['serviceName', 'spanName', 'search', 'minDuration', 'maxDuration', 'limit'];
+
+    // Used to later enforce the type of the key and avoid the error:
+    // "Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'TempoQuery'."
+    const isKeyOfTempoQuery = (key: string): key is keyof TempoQuery => validKeys.includes(key);
+
+    if (query.queryType !== 'nativeSearch') {
+      return query.query ?? '';
     }
-    return query.query ?? '';
+
+    let result: string[] = [];
+    validKeys.forEach((key) => {
+      if (isKeyOfTempoQuery(key) && query[key]) {
```

Comment:
I think if we want to make this typing better this is not ideal. The main point is that the `validKeys` should be part of `TempoQuery` right? We know the `TempoQuery` type in compile time so we should be able to do a compile time error if they don't match at some point. Because in your change if the is a mismatch this will just silently ignore it. 

So you can use type like this:
```
    const keys: Array<
      keyof Pick<TempoQuery, 'serviceName' | 'spanName' | 'search' | 'minDuration' | 'maxDuration' | 'limit'>
    > = ['serviceName', 'spanName', 'search', 'minDuration', 'maxDuration', 'limit'];
```

to specify the keys array and if you make a typo here or something changes in the TempoQuery type you will get a compile time error.

Reply:
I think we had this problem even in the original code, given that we simply had
```
const key of ['serviceName', 'spanName', 'search', 'minDuration', 'maxDuration', 'limit']
```
But I see your point of making this more robust to typos and changes, so I'll update this.

Reply:
Yes previously it also wasn't type safe that much but we shouldn't just remove betterer warning for the sake of it but really make the code more type safe.